lib/tasks/convert.rake

```diff
@@ -66,16 +66,50 @@ namespace :db do
   namespace :convert do
     desc 'Convert/import production data to development.   DANGER Deletes all data in the development database.   Assumes both schemas are already migrated.'
     task :prod2dev => :environment do
+      # dynflow:migrate migrates the db configured for the current rails env
+      # In this case, we need to make sure it migrates development
+      env_bak = ::Rails.env
+      Rake::Task['dynflow:migrate'].invoke
+      ::Rails.env = env_bak
```

Comment:
why do we need to do this? does the dynflow task modify the environment?

Reply:
We need to change the current environment to always migrate the db associated with `development`, regardless of what `::Rails.env` is set to. It felt right to restore the previous value when we don't need it anymore

Reply:
where do we change that? iiuc this runs before the migration script

Reply:
Erm, now it is being done here https://github.com/theforeman/foreman/pull/6962/files#diff-e0bcc8a3c75fbb59b7937982b3fb59caR72